src/executables/util_sa.c

```diff
@@ -1736,11 +1737,22 @@ diagdb (UTIL_FUNCTION_ARG * arg)
 	    {
 	      if (utility_check_class_name (class_name) != NO_ERROR)
 		{
+		  db_shutdown ();
```

Comment:
How about considering the following approach? With this, the next developer won't have to worry about calling db_shutdown().

```
bool need_db_shutdown = false;

...
db_restart ();
need_db_shutdown = true;

...

error_exit:

if (need_db_shutdown == true)
{
    db_shutdown ();
}
```